nomad/structs/config/consul.go

```diff
@@ -180,19 +180,21 @@ func (c *ConsulConfig) ApiConfig() (*consul.Config, error) {
 		if c.VerifySSL != nil {
 			tlsConfig.InsecureSkipVerify = !*c.VerifySSL
 		}
-		tlsClientCfg, err := consul.SetupTLSConfig(&tlsConfig)
-		if err != nil {
-			return nil, fmt.Errorf("error creating tls client config for consul: %v", err)
-		}
-		config.HttpClient.Transport = &http.Transport{
-			TLSClientConfig: tlsClientCfg,
-		}
-	}
-	if c.EnableSSL != nil && !*c.VerifySSL {
-		config.HttpClient.Transport = &http.Transport{
-			TLSClientConfig: &tls.Config{
-				InsecureSkipVerify: true,
-			},
+
+		if tlsConfig.InsecureSkipVerify {
+			config.HttpClient.Transport = &http.Transport{
+				TLSClientConfig: &tls.Config{
+					InsecureSkipVerify: true,
+				},
+			}
+		} else {
+			tlsClientCfg, err := consul.SetupTLSConfig(&tlsConfig)
```

Comment:
Can we just keep the stuff in the `else` block and delete the if/else. Why do we need to special case the `if tlsConfig.InsecureSkipVerify` case when `consul.SetupTLSConfig()` handles it?